internal/service/grafana/workspace_api_key.go

```diff
@@ -0,0 +1,120 @@
+package grafana
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/managedgrafana"
+	"github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func ResourceWorkspaceAPIKey() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceWorkspaceAPIKeyCreate,
+		Read:   schema.Noop,
+		Update: schema.Noop,
+		Delete: resourceWorkspaceAPIKeyDelete,
+
+		Schema: map[string]*schema.Schema{
+			"key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"key_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 100),
+			},
+			"key_role": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(managedgrafana.Role_Values(), false),
+			},
+			"seconds_to_live": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.IntBetween(1, 2592000),
+			},
+			"workspace_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^g-[0-9a-f]{10}$`), "must be a valid workspace id"),
+			},
+		},
+	}
+}
+
+func resourceWorkspaceAPIKeyCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).GrafanaConn
+	wrkspc_id := d.Get("workspace_id").(string)
+	_, err := FindWorkspaceByID(conn, wrkspc_id)
+
+	if err != nil {
+		return fmt.Errorf("error reading Grafana Workspace (%s): %w", d.Id(), err)
+	}
+
+	input := &managedgrafana.CreateWorkspaceApiKeyInput{
+		KeyName:       aws.String(d.Get("key_name").(string)),
+		KeyRole:       aws.String(d.Get("key_role").(string)),
+		SecondsToLive: aws.Int64(int64(d.Get("seconds_to_live").(int))),
+		WorkspaceId:   aws.String(wrkspc_id),
+	}
+
+	log.Printf("[DEBUG] Creating Grafana API Key: %s", input)
+	output, err := conn.CreateWorkspaceApiKey(input)
+
+	if err != nil {
+		return fmt.Errorf("error creating Grafana API Key : %w", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", wrkspc_id, d.Get("key_name").(string)))
+	d.Set("key", output.Key)
+	return nil
+
+}
+
+func resourceWorkspaceAPIKeyDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).GrafanaConn
+
+	log.Printf("[DEBUG] Deleting Grafana Workspace API Key: %s", d.Id())
+
+	wrkspacID, KeyName, er1 := UserParseID(d.Id())
```

Comment:
Some changes here
- `KeyName` should be `keyName`
- `er1` should be just `err`
- The function name used should not have User since this resource does not have a User. Instead, naming it `WorkspaceKeyParseID` (or something similar) would be more intuitive

Reply:
Changed the function name and other comments addressed.

Reply:
Thanks for the changes. Since there were a number of change, we will need the latest acceptance test results updated in the PR description

Reply:
:~/terraform-provider-aws (f-aws_grafana_workspace_api_key) $ make testacc PKG=grafana TESTS=TestAccGrafana_serial/ApiKey
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/grafana/... -v -count 1 -parallel 20 -run='TestAccGrafana_serial/ApiKey'  -timeout 180m
=== RUN   TestAccGrafana_serial
=== RUN   TestAccGrafana_serial/ApiKey
=== RUN   TestAccGrafana_serial/ApiKey/basic
--- PASS: TestAccGrafana_serial (134.54s)
    --- PASS: TestAccGrafana_serial/ApiKey (134.54s)
        --- PASS: TestAccGrafana_serial/ApiKey/basic (134.54s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/grafana    134.618s
:~/terraform-provider-aws (f-aws_grafana_workspace_api_key) $

Reply:
`:~/terraform-provider-aws (f-aws_grafana_workspace_api_key) $ make testacc PKG=grafana TESTS=TestAccGrafana_serial/ApiKey
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/grafana/... -v -count 1 -parallel 20 -run='TestAccGrafana_serial/ApiKey'  -timeout 180m
=== RUN   TestAccGrafana_serial
=== RUN   TestAccGrafana_serial/ApiKey
=== RUN   TestAccGrafana_serial/ApiKey/basic
--- PASS: TestAccGrafana_serial (134.54s)
    --- PASS: TestAccGrafana_serial/ApiKey (134.54s)
        --- PASS: TestAccGrafana_serial/ApiKey/basic (134.54s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/grafana    134.618s
:~/terraform-provider-aws (f-aws_grafana_workspace_api_key) $ `

Reply:
Acc test added as requested.

Reply:
Great, after resolving the failing CI checks, this PR LGTM

Reply:
The CI checks failed as the previous accttest factories need to be replaced with ProtoV5ProviderFactories. Made the change and checked in the code again.